api/v1alpha1/gitrepository_types.go

```diff
@@ -0,0 +1,66 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type GitRepositorySpec struct {
+	Source GitRepositorySource `json:"source,omitempty"`
+	// GiteaURL is the base URL of GitTea server
+	// +kubebuilder:validation:Required
+	GiteaURL string `json:"giteaURL"`
```

Comment:
Can we not infer this? IIUC we have this statically set here: https://github.com/cnoe-io/idpbuilder/blob/main/pkg/controllers/localbuild/resources/gitea/values.yaml#L40, so could we either statically set that or use a shared field which templates this out?

My concern is that, as a user, why should I need to know the gitea url and supply it?

Reply:
That url will always resolve as `127.0.0.1`. I thought we wanted to make this usable for external clusters as well in the future? 

As a user, you don't have to supply this. Creation of GitReposiroty CRs should be handled in the CLI.

Reply:
+1 on what Manabu suggested. this should be an optional field that resolves to localhost if empty.

Also more generically it should say `GitURL`

Reply:
A default sounds great!